src/Twig/Extension/CompatExtension.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Bolt\Twig\Extension;
+
+use Bolt\Twig\Runtime;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFilter;
+
+/**
+ * Compatibility between storage layers Twig extension.
+ *
+ * @internal
+ *
+ * @author Ross Riley <riley.ross@gmail.com>
+ */
+class CompatExtension extends AbstractExtension
+{
+
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFilters()
+    {
+        $safe = ['is_safe' => ['html']];
+        $deprecated = ['deprecated' => true];
+
+        return [
+            new TwigFilter('taxonomy_compat', [Runtime\CompatRuntime::class, 'taxonomyCompat'], $safe + $deprecated)
```

Comment:
OK, so digging I have a couple of question:

1. Why not just `|taxonomy`
2. I know with 1. you've marked it up as deprecated, but is it, and what is the way forward?
3. Shouldn't this just go into the `RecordRuntime`

Reply:
I might still have a rethink about how to do this better, but yes, the problem it solves is that calling `record.taxonomy` on the legacy content objects gives a very specific key/value array structure that it's impossible to patch in without ruining the concept of having proper collections for taxonomies.

But at the same time, the taxonomy_links twig templates depend on them being this way, so I can't make it BC, so the next best option as far as I can think is to provide a filter to standardise until a time the theme author can rewrite the Twig.

So for this reason, it's new, but immediately deprecated so that we can remove in 4.0 by which time themes can have been tweaked to use the new structure.

Finally, I just thought it was better not to add the legacy config stuff into existing extensions but I don't mind either way, I started doing it in RecordRuntime and just thought it might be more isolated as a separate extension that we can throw away as soon as it's time has passed.

Reply:
Oh totally in favour of a filter as opposed to trying to make it work as a method on the object, and that was the closest consensus we got to in previous meetings. Deity classes :fire: :+1: 

I was more thinking the naming of the filter, as people are getting increasingly narky at changes we make … that single point is my worry here. I know we are approaching this from "It is deprecated and will change", but I am just wondering after a few weeks of various people getting upset, it might be good to think about how we give them the way it will be to transition to, rather than one change now, and another at the end of the year … I don't have a solution either by the way, just trying to get the conversation happening to take some of this off you and I.

But if I am totally missing the point, please throw a wet rag at me.